nideep/datasets/amfed/amfed.py

```diff
@@ -93,23 +100,150 @@ def find_data_intersection(self):
         if paths_au_labels is not None and len(paths_landmarks) > 0:
             lists.append(list_comparables(paths_landmarks, suffix='-landmarks'))
             list_names.append(AMFED.KEY_LANDMARKS)
-        
+
         common_entity_names = list(set(lists[0]).intersection(*lists[1:]))
-        common_entities = self.from_list_to_typle(common_entity_names, lists, list_names)
-        self.entities = common_entities
-        return common_entities
-    
+        common_entities = self.from_list_to_tuple(common_entity_names, lists, list_names)
+        return pd.DataFrame([c._asdict() for c in common_entities])
+
+    def as_dataframe(self):
+        index_path = self.cache_dir + AMFED.INDEX_FILE
+        if os.path.isfile(index_path):
+            result_df = pd.read_csv(index_path)
+        else:
+            entities = self.find_data_intersection()
+            count = 0
+            result = []
+            for _, e in entities.iterrows():
+                entity_records = self.load_file(e)
+                result += entity_records
+                print 'Finished video %d/%d' % (count + 1, len(entities))
+                count += 1
+
+            result_df = pd.DataFrame(result)
+            result_df.to_csv(index_path)
+
+        return result_df
+
+    def as_numpy_array(self):
+        df = self.as_dataframe()
+        df = df[df.valid]
+        # X_train = np.array([cv2.imread(e, cv2.IMREAD_GRAYSCALE) for e in df['features'].tolist()])
+        X_train = np.array([cv2.imread(e) for e in df['features'].tolist()])
+        y_train = np.array([pickle.load(open(e, 'rb')) for e in df['labels'].tolist()])
+        return X_train, y_train
+
+    def load_file(self, entity):
+        """
+        Loads a specific video of the dataset, preprocess it (frame extraction, resizing) and return it as a numpy array
+        :return: A list of dictionaries
+        """
+        labels_df = pd.read_csv(entity.au_label)
+        landmarks_df = pd.read_csv(entity.landmarks, header=None)
+
+        vidcap = cv2.VideoCapture(entity.video)
+        success, image = vidcap.read()
+        count = 0
+        success = True
```

Comment:
You're overwriting the value of success returned by the first call to vidcap.read() but you don't need need to and probably shouldn't either.